spec/unit/provider/group/dscl_spec.rb

```diff
@@ -117,20 +117,27 @@
     before do
       @node = Chef::Node.new
       @provider = Chef::Provider::Group::Dscl.new(@node, @new_resource)
-      allow(@provider).to receive(:safe_dscl).and_return("\naj      500\n")
     end
 
-    it "should run safe_dscl with list /Groups gid" do
-      expect(@provider).to receive(:safe_dscl).with(*"list /Groups gid".split(" "))
+    it "should run safe_dscl with search /Groups gid" do
+      expect(@provider).to receive(:safe_dscl).with(*"search /Groups gid 500".split(" ")).and_return('')
       @provider.gid_used?(500)
     end
 
     it "should return true for a used gid number" do
+      allow(@provider).to receive(:safe_dscl).and_return(<<-eos
```

Comment:
put this in the before block so that your other tests are actually useful...

Reply:
I don't quite follow how having this in the before block is helpful. The check for whether the given gid is used is whether safe_dscl returns any output or not. For the tests of truthy/falsey values it would have to mock different things.